Kernel/FileSystem/BlockBasedFileSystem.cpp

```diff
@@ -174,6 +171,29 @@ int BlockBasedFS::write_block(unsigned index, const UserOrKernelBuffer& data, si
     return 0;
 }
 
+bool BlockBasedFS::force_cache_block(unsigned index) const
+{
+    auto& entry = cache().get(index);
+    u32 base_offset = static_cast<u32>(index) * static_cast<u32>(block_size());
+    file_description().seek(base_offset, SEEK_SET);
+    auto entry_data_buffer = UserOrKernelBuffer::for_kernel_buffer(entry.data);
+    auto nread = file_description().read(entry_data_buffer, block_size());
+    if (nread.is_error())
+        return false;
+    ASSERT(nread.value() == block_size());
+    entry.has_data = true;
+    return true;
+}
+
+BlockBasedFS::CacheEntry& BlockBasedFS::cache_block(unsigned index) const
+{
+    auto& entry = cache().get(index);
+    if (!entry.has_data) {
+        force_cache_block(index);
```

Comment:
We're forgetting to check the return value of `force_cache_block` here. If there's an error we need to handle it.

Reply:
Oh right, if we can't cache here I'd assume that we should assert.

Reply:
Hmm.. if this is dealing with hardware, then hardware can fail at any time, so we need to handle it more gracefully than just asserting I think

Reply:
I think I figured this out, I use `KResultOr` to indicate a failure.